sqlglot/parser.py

```diff
@@ -747,6 +747,9 @@ class Parser(metaclass=_Parser):
 
     CONVERT_TYPE_FIRST = False
 
+    QUOTED_PIVOT_COLUMNS: t.Optional[bool] = None
```

Comment:
why is this optional?

Reply:
It's because we use its value as an argument to `exp.to_identifier` and it has different semantics for `False` and `None`. We must quote if the identifier is not safe in dialects that create names such as `pivotcolumn_avg(col)`.